common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -428,4 +433,21 @@ public synchronized void sessionHeartbeat() throws IOException {
       throw new IOException(e);
     }
   }
+
+  /**
+   * Called only by {@link WorkerClientHeartbeatExecutor}, encapsulates
+   * {@link #sessionHeartbeat()} in order to cancel and cleanup the
+   * heartbeating thread in case of failures
+   */
+  public synchronized void periodicHeartbeat() {
```

Comment:
sessionHeartbeat() is also called directly by BlockWorkerClientPool. Adding the new method to define logic which is only specific to the periodic HB thread.